src/core/GameKeyHandler.ttslua

```diff
@@ -57,6 +58,80 @@ function addDoomToAgenda()
   doomCounter.call("addVal", 1)
 end
 
+-- move the hovered object to the nearest empty slot on the playermat
+function takeCardIntoThreatArea(playerColor, hoveredObject)
+  -- only continue if an unlocked card
+  if hoveredObject == nil
+      or hoveredObject.type ~= "Card"
+      or hoveredObject.hasTag("Location")
+      or hoveredObject.locked then
+    broadcastToColor("Hover a non-location and try again.", playerColor, "Yellow")
+    return
+  end
+
+  local matColor = playmatApi.getMatColor(playerColor)
+  local mat = guidReferenceApi.getObjectByOwnerAndType(matColor, "Playermat")
+  -- do not continue if the threat area is already full
+  if playmatApi.getEncounterCardDrawPosition(matColor, false) == playmatApi.getEncounterCardDrawPosition(matColor, true) then 
```

Comment:
I think you actually need to compare the coordinates since Lua does only check if they are the same table reference if you do it like this (and that will always be false).

Reply:
Nvm this works